src/field-builder.ts

```diff
@@ -145,3 +146,59 @@ fieldBuilderProto.effect = function effect({ effect = {}, resolve, ...options })
     }) as Resolver<any, any, any, any>,
   });
 };
+
+fieldBuilderProto.effectConnection = function connection(
+  { edgesNullable, nodeNullable, type, ...fieldOptions },
+  connectionOptionsOrRef = {} as never,
+  edgeOptionsOrRef = {} as never,
+) {
+  const connectionRef = connectionOptionsOrRef instanceof ObjectRef
+    ? connectionOptionsOrRef
+    : this.builder.objectRef<ConnectionShape<SchemaTypes, unknown, boolean>>(
+      'Unnamed connection',
+    );
+
+  const fieldRef = this.effect({
+    ...this.builder.options.relayOptions?.defaultConnectionFieldOptions,
+    ...fieldOptions,
+    args: {
+      ...fieldOptions.args,
+      ...this.arg.connectionArgs(),
+    },
+    resolve: fieldOptions.resolve as never,
+    type: connectionRef,
+  } as never);
+
+  if (!(connectionOptionsOrRef instanceof ObjectRef)) {
+    this.builder.configStore.onFieldUse(fieldRef, (fieldConfig) => {
+      const connectionName = connectionOptionsOrRef.name
+        ?? `${this.typename}${capitalize(fieldConfig.name)}${
+          fieldConfig.name.toLowerCase().endsWith('connection') ? '' : 'Connection'
+        }`;
+
+      this.builder.connectionObject(
+        {
+          edgesNullable,
+          nodeNullable,
+          type,
+          ...connectionOptionsOrRef,
+          name: connectionName,
+        },
+        edgeOptionsOrRef instanceof ObjectRef
+          ? edgeOptionsOrRef
+          : {
+            name: `${connectionName}Edge`,
+            ...edgeOptionsOrRef,
+          },
+      );
+
+      this.builder.configStore.associateRefWithName(connectionRef, connectionName);
```

Comment:
yes, this is basically saying that the connectionRef should resolve to the type associated with connectionName.  Internals get a little complicated due to backwards compatibility and historic reasons, but the implementations for types are stored in a map of name -> implementation, refs are stored as ref -> name, so to resolve a ref, the lookup is something like ref -> name -> implementation.  This is a little weird because you need to delay the creation of the object type until after you have the name of the field to generate a type name that matches the field name.  Normally you wouldn't need to use this method because you can just pass the ref into a method like builder.objectType